tests/Feature/ProductDeleteTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Models\Product;
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Tests\TestCase;
+
+class ProductDeleteTest extends TestCase
+{
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->artisan('migrate:fresh');
+        $this->artisan('db:seed');
+    }
+
+    public function test_auth(): void
+    {
+        $admin = User::where('email', 'admin@temu.com')->first();
+        $user = User::where('email', 'user@temu.com')->first();
+
+        $product = Product::all()->random();
+
+        # Those without admin role cannot delete
+        $this->actingAs($user)
+            ->delete(route('api.products.images.delete', [$product->id, $product->images->first()->id]))
+            ->assertStatus(403);
+
+        # Those with admin role can delete
+        $this->actingAs($admin)
```

Comment:
Very well structured tests , fetching the products and testing the policy for user authentication .
But should it be a test for not logged in user as well?

Reply:
Good point! Added unauthed test bellow: https://github.com/VaguelyOnline/laravel-vue-temu/pull/15/commits/33aac78f54da5fc7c24906373cf8caaa5385edcf